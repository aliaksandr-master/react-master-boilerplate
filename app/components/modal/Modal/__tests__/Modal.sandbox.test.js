/* eslint-env browser, jest */
/* eslint-disable react/jsx-no-bind */

import React from 'react';
import renderer from 'react-test-renderer';
import { wrapSandbox } from '../../../../vendor/react-sandbox/sandbox-staging';
import sandboxModal from '../Modal.sandbox';
import Test from '../../../test/Test';



describe('rendering', () => {
  wrapSandbox(sandboxModal).forEach(({ name, template, props }) => {
    it(`should render sandbox "${name}"`, () => {
      const Modal = template();

      const component = renderer.create(
        <Test>
          <Modal {...props} />
        </Test>
      );

      const tree = component.toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
