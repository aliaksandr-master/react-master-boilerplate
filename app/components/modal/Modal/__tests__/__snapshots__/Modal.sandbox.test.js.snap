// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render sandbox "default" 1`] = `
<div>
  <button
    onClick={[Function]}
    type="button"
  >
    Show Modal
  </button>
  <div>
    <h1>
      REACT_MODAL
    </h1>
    <p>
      {
    "className": "Modal Modal--theme-default",
    "closeTimeoutMS": 210,
    "isOpen": true,
    "shouldCloseOnOverlayClick": true,
    "contentLabel": "Modal"
}
    </p>
    <div>
      <button
        className="Modal__close"
        onClick={[Function]}
        type="button"
      >
        Ã—
      </button>
      <div
        className="Modal__in"
        style={
          Object {
            "width": 500,
          }
        }
      >
        Hello world
      </div>
    </div>
  </div>
</div>
`;
